resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -94,6 +94,14 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private var totalCoresAcquired = 0
   private var totalGpusAcquired = 0
 
+  // The amount of time to wait for locality scheduling
+  private val localityWait = conf.getTimeAsMs("spark.locality.wait", "3s")
```

Comment:
Can you instead add spark.locality.wait to https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/config/package.scala and use the `ConfigEntry` for it?